### PR TITLE
Closes #2829 All external links opening in new tab instead of only intended external links

### LINKED ISSUE #2829: All external links opening in new tab instead of only intended external links
## Problem/Motivation
The 2.8 release includes functionality to open certain kinds of external links, such as documents and Trellis Events in a new tab by default. This is working, but it is also causing all other external links to open in a new tab, which is not supposed to happen.

**Example test links**

- [Trellis event link](https://events.trellis.arizona.edu/f44lNu67/5a3U3nIwLYK?_gl=1*1ncy4ls*_ga*OTMyMDU0ODAzLjE2OTE0MjM0NjM.*_ga_7PV3540XS3*MTY5NTY3MDQ4NC4zNS4xLjE2OTU2NzA0OTUuNDkuMC4w)
- [Google doc](https://docs.google.com/document/d/1hYZIm553E7gO_bCxr1K7l0NN_v137k6jNlkUyyzCroY/edit?usp=sharing)
- [Box link](https://arizona.box.com/s/wasg24vebx58n5sv13n9zg68aox59bh0)
- [OneDrive link](https://emailarizona-my.sharepoint.com/:x:/g/personal/danahertzberg_arizona_edu/EcyTb-u8uGpHnmeogdAYRqUBnVZjOPKMxX5v5OB6XuA6Sw?e=KTFRGx)
- [Dropbox](https://www.dropbox.com/scl/fi/8wcc4shset2qkbt3qc6wh/21-UArizona-Dynamic-Template.pptx?rlkey=ms6028rbz9m2nrje7n907o9fg&dl=0)
- [Sample Document.pdf](https://github.com/az-digital/az_quickstart/files/12894683/Sample.Document.pdf)

**Here’s an updated list of all the beta testing sites with links to their multidevs:**

- :white_check_mark: azs-quickstart: [qs-2825-azs-quickstart.pantheonsite.io/dana](http://qs-2825-azs-quickstart.pantheonsite.io/dana)
   - All external links open in new tab
- :white_check_mark: azs-capla: [azqs28-azs-capla.pantheonsite.io/dana](http://azqs28-azs-capla.pantheonsite.io/dana)
   - All external links open in new tab
- :white_check_mark: sasg-rec: https://pre-release-sasg-rec.pantheonsite.io/dana
   - All external links open in new tab
- :white_check_mark: azs-law: https://azqs28-azs-law.pantheonsite.io/dana
   - All external links open in a new tab
   - Messing with extlink settings on this site
- :white_check_mark: azs-eller: [pre-release-azs-eller.pantheonsite.io](http://pre-release-azs-eller.pantheonsite.io/)
- :white_check_mark: sasg-financialaid: [pre-release-sasg-financialaid.pantheonsite.io](http://pre-release-sasg-financialaid.pantheonsite.io/)
- :white_check_mark: azs-thinktank: [pre-release-azs-thinktank.pantheonsite.io](http://pre-release-azs-thinktank.pantheonsite.io/)
- :white_check_mark: sasg-registrar: [azqs28-sasg-registrar.pantheonsite.io/](http://azqs28-sasg-registrar.pantheonsite.io/)
- :white_check_mark: azs-cert: [pre-release-azs-cert.pantheonsite.io/](http://pre-release-azs-cert.pantheonsite.io/)
- :white_check_mark: azs-crm: [28-beta1-azs-crm.pantheonsite.io](http://28-beta1-azs-crm.pantheonsite.io/)

### PR DESCRIPTION
**Pull request by bberndt-uaz**
## Description

Updates the settings for the External Links module to remove the list of whitelisted domains and instead use a regular expression to open external URLs in a new tab/window only if they start with one of the following:

- https://arizona.box.com
- https://docs.google.com
- https://dropbox.com
- https://www.dropbox.com
- https://emailarizona-my.sharepoint.com
- https://events.trellis.arizona.edu

## Related issues

#2829

## How to test

Link to Probo site:

Confirm that the only links to open in a new tab/window are links to PDFs on the website and the external URLs listed above.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by joeparsons — APPROVED** [review]

File: config/install/extlink.settings.yml [hunk 1]
```diff
@@ -13,16 +13,11 @@ extlink_include: '/sites/.+/files/.+\.pdf'
 extlink_class: '0'
 extlink_label: '(link is external)'
 extlink_img_class: false
-extlink_css_exclude: ''
+extlink_css_exclude: '^(?!.*(events.trellis.arizona.edu|arizona.box.com|docs.google.com|dropbox.com|emailarizona-my.sharepoint.com))'
```
Comment (bberndt-uaz) [inline config/install/extlink.settings.yml:16]:
```suggestion
extlink_css_exclude: '^(?!(https://)(arizona.box.com|docs.google.com|dropbox.com|www.dropbox.com|emailarizona-my.sharepoint.com|events.trellis.arizona.edu))'
```

**Review by joeparsons — APPROVED** [review]

**Comment by joeparsons** [comment]
- Updated regex pattern slightly (to escape slashes and account for content after domain in URLs)
- Moved updated regex to correct config setting
- Re-exported `extlink.settings.yml`

**Review by joeparsons — APPROVED** [review]

**Comment by joeparsons** [comment]
Updated regex can be tested here:
https://regex101.com/r/Iy5jiZ/1

**Comment by danahertzberg** [comment]
Will there still be a db update included from the last PR?

**Review by danahertzberg — APPROVED** [review]
Working as expected! 
https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2833.probo.build/dana

**Review by tadean — APPROVED** [review]

**Pull request merged**